extractmanager.py

```diff
@@ -0,0 +1,105 @@
+import sys
+import re
+import os
+from pathlib import Path
+import shutil
+
+import disc_riider_py
+
+WIT_PROGRESS_REGEX = re.compile(rb" +([0-9]+)%.*")
+CLEAN_NTSC_U_1_00_DOL_HASH = "450a6806f46d59dcf8278db08e06f94865a4b18a"
+
+WRONG_VERSION_DOL_HASHES = {
+    "2848bb574bfcbf97f075adc4e0f4692ddd7fd0e8": "JP 1.00",
+    # "TODO": "US 1.01",
+    "30cad7e8a88442b1388867f01bc6461097f4a152": "US 1.02",
+    "8f6bf468447d9f10172cc4a472a56e1f526a5cb4": "PAL 1.00",
+    # "TODO": "PAL 1.01",
+    # "TODO": "PAL 1.02",
+}
+
+NOP = lambda *args, **kwargs: None
+
+# currently, only win and linux (both 64bit) are supported
+IS_WINDOWS = sys.platform == "win32"
+
+
+class WrongChecksumException(Exception):
+    pass
+
+
+class ExtractManager:
+    def __init__(self, rootpath: Path):
+        self.rootpath = rootpath
+
+    def actual_extract_already_exists(self):
+        return (
+            self.rootpath / "actual-extract" / "DATA" / "sys" / "main.dol"
+        ).is_file()
+
+    def extract_game(self, iso_path, progress_cb=NOP):
+        if not self.actual_extract_already_exists():
+            dest_path = self.rootpath / "actual-extract"
+            extractor = disc_riider_py.WiiIsoExtractor(iso_path)
+            extractor.prepare_extract_section("DATA")
+            # remove hint videos, but keep credits
+            extractor.remove_files_by_callback(
+                "DATA", lambda x: x.startswith("THP") and not "Demo" in x
+            )
+            checksum = bytes(extractor.get_dol_hash("DATA")).hex()
+            if "450a6806f46d59dcf8278db08e06f94865a4b18a" != checksum:
```

Comment:
```suggestion
            if CLEAN_NTSC_U_1_00_DOL_HASH != checksum:
```
You declare the constant but don't actually use it :p